code/modules/research/protolathe.dm

```diff
@@ -95,3 +96,58 @@ Note: Must be placed west/left of and R&D console to function.
 			use_power(max(1000, (MINERAL_MATERIAL_AMOUNT * M.last_amount_inserted / 10)))
 		add_overlay("protolathe_[stack_name]")
 		addtimer(CALLBACK(src, /atom/proc/cut_overlay, "protolathe_[stack_name]"), 10)
+
+/obj/machinery/rnd/protolathe/proc/user_try_print_id(id, amount)
+	if(!istype(linked_console) || !id)
+		return FALSE
+	if(isnull(amount))
+		amount = 1
+	if(istext(amount))
+		amount = text2num(amount)
+	var/datum/design/D = linked_console.stored_research.researched_designs[id]
+	if(!istype(D))
+		return FALSE
+	if(D.make_reagents.len)
+		return FALSE
+
+	var/power = 1000
+	amount = Clamp(amount, 1, 10)
+	for(var/M in D.materials)
+		power += round(D.materials[M] * amount / 5)
+	power = max(3000, power)
+	use_power(power)
+
+	var/list/efficient_mats = list()
+	for(var/MAT in D.materials)
+		efficient_mats[MAT] = D.materials[MAT]*efficiency_coeff
+
+	if(!materials.has_materials(efficient_mats, amount))
+		say("Not enough materials to complete prototype[amount > 1? "s" : ""].")
+		return FALSE
+	for(var/R in D.reagents_list)
+		if(!reagents.has_reagent(R, D.reagents_list[R]*efficiency_coeff))
+			say("Not enough reagents to complete prototype[amount > 1? "s" : ""].")
+			return FALSE
+
+	materials.use_amount(efficient_mats, amount)
+	for(var/R in D.reagents_list)
+		reagents.remove_reagent(R, D.reagents_list[R]*efficiency_coeff)
+
+	busy = TRUE
+	flick("protolathe_n", src)
+	var/timecoeff = efficiency_coeff * D.lathe_time_factor
+
+	addtimer(CALLBACK(src, .proc/reset_busy), (32 * timecoeff * amount) ** 0.8)

```

Comment:
What's with the weird scaling here.

Reply:
this was what it was at previously so this is what i put it at now